.github/workflows/CI.yaml

```diff
@@ -25,7 +25,15 @@ jobs:
       fail-fast: false
       matrix:
         os: [macOS-latest, ubuntu-latest, windows-latest]
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        include:
+          - name: numpy_min
+            os: ubuntu-latest
+            python-version: 3.6
+            run_type: FULL
+            install_hole: false
+            codecov: false
+            numpy: numpy=1.16.0
```

Comment:
Have you checked this test to see if it's actually installing 1.16.0 (and paying attention to the `codecov` and `python-version` flags? My hunch is not necessarily, because the workflow installs dependencies from a yaml file instead of installing numpy manually. In MDAnalysis this flag works because of [these lines](https://github.com/MDAnalysis/mdanalysis/blob/8f9f402fe65d6a90e88b60b6da735c130425f00e/.github/workflows/gh-ci.yaml#L133):

```
conda_deps="${{ matrix.numpy }} ${MDA_CONDA_${{ matrix.run_type }}_DEPS}"
mamba install ${conda_deps}
```

which basically expand out to

```
conda_deps="numpy=1.16.0 scipy pytest ..."
mamba install numpy=1.16.0 scipy pytest ...
```

([mamba](https://github.com/mamba-org/mamba) is a drop-in replacement for conda).

I'm not sure if this is the most elegant solution (@IAlibay knows GH Actions much better than me), but you could have something like an optional numpy re-installation step:

```
- name: reinstall_numpy
- shell: bash
- run: |
  if [[ $numpy ]] ; then  # if the $numpy bash variable is set
  	conda -y uninstall numpy  # uninstall numpy without asking for confirmation
  	conda -y install $numpy  # install numpy version without asking for confirmation
  done
```

I can't remember if conda confirmation is necessary in GH actions.

Reply:
Thanks. I checked with Python 3.7 and confirmed I got `numpy=1.16.0`. I didn't check for 3.6 because I took it for granted and now that I checked I saw `numpy=1.19.5`. For Python 3.8 and higher, I am getting `numpy=1.21.1` (reason why I also added the `1.20.2` version in the first submission of the PR.)
But then there is something I am still missing. If the workflow installs dependencies from the yaml file, as you said, why does MDAnalyis have >=1.17.3 in [the yaml file](https://github.com/MDAnalysis/mdanalysis/blob/develop/maintainer/conda/environment.yml), and [the CI](https://github.com/MDAnalysis/mdanalysis/blob/develop/.github/workflows/gh-ci.yaml) has the `min_numpy` with 1.16.0?